std/experimental/allocator/building_blocks/allocator_list.d

```diff
@@ -632,119 +637,340 @@ version(Posix) @system unittest
     assert(b1.length == 1024 * 10);
 }
 
+/// Ditto
+shared struct SharedAllocatorList(Factory, BookkeepingAllocator = GCAllocator)
+{
+    import std.typecons : Ternary;
+    import std.traits : hasMember;
+    import core.internal.spinlock : SpinLock;
+
+private:
+    // Forward all calls to 'impl' and protect them by the lock below
+    AllocatorList!(Factory, BookkeepingAllocator) impl;
+    SpinLock lock = SpinLock(SpinLock.Contention.brief);
+
+    // Remove this when 'https://github.com/dlang/druntime/pull/2156' gets merged
+    static ref T assumeUnshared(T)(ref shared T val) @trusted @nogc pure nothrow
+    {
+        return *cast(T*) &val;
+    }
+
+    // Debug function used for testing
+    version(unittest)
+    auto allocators()
+    {
+        return impl.allocators;
+    }
+
+// Everything is inherited from the 'AllocatorList' implementation
+public:
+    /**
+    The alignment offered.
+    */
+    enum alignment = impl.alignment;
+
```

Comment:
That `static foreach` got implemented, then reverted later.
https://github.com/dlang/phobos/compare/27fa9abb91cd563075958411c00817245e7b6c15..f91e9e7e77b3f3e023ab96241389636d12cef03d

Reply:
Given this PR already had an “auto-merge” label after that reverting, I’d say, we keep it as-is for now. In case someone wants to opt for the clever meta programming loop, they are free to send their PR.